components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/service/EventSimulator.java

```diff
@@ -119,6 +145,35 @@ public static void validateSimulationConfig(String simulationConfiguration) thro
                 throw new InvalidConfigException("Simulation properties are required for event simulation. Invalid " +
                         "simulation configuration provided : " + simulationConfig.toString());
             }
+        } catch (ResourceNotFoundException e) {
```

Comment:
we need to check whether the ResourceNotFoundException is the same as the previously thrown exception, if the exceptions are different we need to update the inactiveSimulations map and log an error. The condition check ensures that we don't log errors when the ResourceNotFound exception is thrown for the same resource. 

Is there a better way to do the same?